modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx

```diff
@@ -0,0 +1,192 @@
+import ClayModal from '@clayui/modal';
+import {FieldBase} from 'frontend-js-components-web';
+import {fetch, navigate, objectToFormData} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+type FrontendTokenDefinitionProvider = {
+	name: string;
+	themeId: string;
+};
+
+interface AddStyleBookModalProps {
+	addStyleBookEntryURL: string;
+	closeModal: () => void;
+	frontendTokenDefinitionProviders?: Array<FrontendTokenDefinitionProvider>;
+	namespace: string;
+}
+
+const AddStyleBookModalContent = ({
+	addStyleBookEntryURL,
+	closeModal,
+	frontendTokenDefinitionProviders = [],
+	namespace,
+}: AddStyleBookModalProps) => {
+
+	const [loading, setLoading] = useState(false);
+	const [errorMessage, setErrorMessage] = useState<string>('');
+	const [name, setName] = useState<string>('');
+	const [themeId, setThemeId] =
+		useState<any>(frontendTokenDefinitionProviders[0].themeId);
+
+	const handleFormError = (responseContent: any) => {
+		setErrorMessage(responseContent.error || '');
+	};
+
+	const handleSubmit = (event: any) => {
+		event.preventDefault();
+
+		const error = name.trim().length
+			? ''
+			: Liferay.Language.get('this-field-is-required');
+
+		if (error) {
+			setErrorMessage(error);
+
+			return;
+		}
+
+		setLoading(true);
+
+		const body = Liferay.Util.ns(namespace, {
+			name,
+			themeId: themeId.themeId,
+		});
+
+		fetch(addStyleBookEntryURL, {
+			body: objectToFormData(body),
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((responseContent) => {
+				if (responseContent.error) {
+					setLoading(false);
+					setErrorMessage(responseContent.error);
+				}
+				else if (responseContent.redirectURL) {
+					navigate(responseContent.redirectURL, {
+						beforeScreenFlip: closeModal,
+					});
+				}
+			})
+			.catch((response) =>
+				handleFormError(response)
+			);
+	};
+
+	const nameId = `${namespace}name`;
+	const themeIdId = `${namespace}tokenDefinition`;
+	const saveButtonId = `${namespace}saveButton`;
+
+	return (
+		<>
+			<ClayModal.Header>
+				{Liferay.Language.get('add-style-book')}
+			</ClayModal.Header>
+			<ClayModal.Body>
+
+				<ClayForm id={saveButtonId} onSubmit={handleSubmit}>
+
+					<FieldBase
+						className="themeId"
+						helpMessage={Liferay.Language.get(
+							'the-style-book-will-be-created-based-on-the-selected-token-definition'
+						)}
+						id={themeIdId}
+						label={Liferay.Language.get('create-style-book-for')}
+					>
+						<Picker
+							defaultSelectedKey={themeId}
+							id={themeIdId}
+							onSelectionChange={setThemeId}
+							selectedKey={themeId.themeId}
+						>
+							{frontendTokenDefinitionProviders.map(
+								(frontendTokenDefinitionProvider) => (
+									<Option
+										key={frontendTokenDefinitionProvider.themeId}
+										textValue={frontendTokenDefinitionProvider.name}
+									>
+										{frontendTokenDefinitionProvider.name}
+									</Option>
+								)
+							)}
+						</Picker>
```

Comment:
Instead of rendering the options in a map directly as a child of the Picker, it would be better to pass the items to the Picker so that you can have more benefits from the Picker being able to control and cache when necessary for optimizations and adjustments on mobile as well. This makes the component dynamically rendered instead of static.

```jsx
<Picker
	defaultSelectedKey={themeId}
	id={themeIdId}
	onSelectionChange={setThemeId}
	selectedKey={themeId.themeId}
	items={frontendTokenDefinitionProviders}
>
    {(item) => (
        <Option
           	key={item.themeId}
           	textValue={item.name}
        >
       	    {item.name}
        </Option>
    )}
</Picker>
```